File: Cite.EvalIt/Query/DataObjectLookup.cs
using Cite.EvalIt.Common;
using Cite.Tools.Data.Query;
using System;
using System.Collections.Generic;
using System.Text;

namespace Cite.EvalIt.Query
{
	public class DataObjectLookup : Lookup
	{
		public string Like { get; set; }
		public string LikeDescription { get; set; }
		public List<Guid> Ids { get; set; }
		public List<Guid> ExcludedIds { get; set; }
		public List<Guid> TagIds { get; set; }
		public List<Guid> TypeIds { get; set; }
		public List<Guid> UserIds { get; set; }
		public List<IsActive> IsActive { get; set; }

		//public UserQuery Enrich(QueryFactory factory)
		//{
		//	UserQuery query = factory.Query<UserQuery>();

		//	if (this.Ids != null) query.Ids(this.Ids);
		//	if (this.ExcludedIds != null) query.ExcludedIds(this.ExcludedIds);
		//	if (this.IsActive != null) query.IsActive(this.IsActive);

		//	this.EnrichCommon(query);

		//	return query;
		//}
	}
}

File: Cite.EvalIt.WebApp/src/app/core/query/data-object.lookup.ts
import { IsActive } from '@app/core/enum/is-active.enum';
import { Lookup } from '@common/model/lookup';
import { Guid } from '@common/types/guid';

export class DataObjectLookup extends Lookup implements DataObjectFilter {
	ids: Guid[];
	excludedIds: Guid[];
	tagIds: Guid[];
	typeIds: Guid[];
	userIds: Guid[];
	isActive: IsActive[];
	like: string;
	likeDescription: string;

	constructor() {
		super();
	}
}

export interface DataObjectFilter {
	ids: Guid[];
	excludedIds: Guid[];
	tagIds: Guid[];
	typeIds: Guid[];
	userIds: Guid[];
	isActive: IsActive[];
	like: string;
	likeDescription: string;
}